plugins/org.eclipse.sirius.diagram.sequence/src/org/eclipse/sirius/diagram/sequence/tool/internal/validation/description/constraints/UniqueInteractionContainerMappingConstraints.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2024 CEA.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.diagram.sequence.tool.internal.validation.description.constraints;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.validation.AbstractModelConstraint;
+import org.eclipse.emf.validation.IValidationContext;
+import org.eclipse.sirius.diagram.description.Layer;
+import org.eclipse.sirius.diagram.sequence.description.InteractionContainerMapping;
+import org.eclipse.sirius.diagram.sequence.description.SequenceDiagramDescription;
+
+/**
+ * Validation rule checking that only one Interaction Container Mapping is present in a VSM.
+ * 
+ * @author smonnier
+ */
+public class UniqueInteractionContainerMappingConstraints extends AbstractModelConstraint {
+
+    /**
+     * Sequence children presentation rule id.
+     */
+    public static final String UNIQUE_INTERACTION_CONTAINER_MAPPING_ID = "org.eclipse.sirius.diagram.sequence.UniqueInteractionContainerMapping"; //$NON-NLS-1$
+
+    @Override
+    public IStatus validate(IValidationContext ctx) {
+        IStatus result = ctx.createSuccessStatus();
+        final EObject eObj = ctx.getTarget();
+        if (UNIQUE_INTERACTION_CONTAINER_MAPPING_ID.equals(ctx.getCurrentConstraintId())) {
+            if (eObj instanceof SequenceDiagramDescription sequenceDiagramDescription) {
```

Comment:
The validation works only if we launch it from the SequenceDIagramDescription. If it is launched from a layer for instance, it does not verify the rule

Reply:
as other validation rules : it seems that there would be several marker if we to this from layer 
We need to compute it from the Sequence diagram description in all cases.
(Even if it will not get layers from DiagramExtensionDescription).

Reply:
Yes, at first I filtered InteractionContainerMapping instead of SequenceDiagramDescription, but that would duplicate for each mapping so I changed it to be on the diagram level.

Reply:
New version of the validation rule.
Will produce two markers if a layer contains to IC mappings: one of the layer, one on the diagram